File: factorial_iterative.rb
def factorial(number)
  factorial_iterative(1,1,number)
end

def factorial_iterative(product,accumulator,max_count)
  if accumulator > max_count
    return product
  else
    factorial_iterative((product*accumulator),(accumulator+1),(max_count))
  end
end

p factorial(5)

File: factorial_iterative.js
#!/usr/bin/env node
// factorial iterative

function factorial(n) {
  return factorialIter(1,1,n);
}

function factorialIter(product,counter,max) {
  if (counter > max) {
    return product;
  } else {
    return factorialIter((product*counter),(counter+1),max)
  }
}

console.log(factorial(19))
